env_setup/env_cleanup.yml

```diff
@@ -39,3 +39,24 @@
       when:
         - cleanup_vm | bool
         - new_vm is defined and new_vm | bool
+
+    - name: "Restore base snapshot for windows update installation"
+      when:
+        - win_update_snapshot_name is defined
+        - win_update_snapshot_name
+        - (vm_exists is defined and vm_exists)
```

Comment:
If the VM is cleaned in above tasks, no need to execute tasks in this block. So you can set `vm_exists` to false together with setting `vm_cleaned_up: true`.